lib/customer.rb

```diff
@@ -0,0 +1,47 @@
+require 'csv'
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  @@all_customers = []
+
+  def initialize(id, email, address = {})
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.customer_data
+    if @@all_customers.length != 0
+      return
+    end
```

Comment:
Such a clever way to prevent the program from needing to do the same work over and over!